client/workflow.go

```diff
@@ -0,0 +1,271 @@
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.PurgeWorkflowBeta1(ctx, &pb.PurgeWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to purge workflow: %v", err)
+	}
+	return nil
+}
+
+// TerminateWorkflowBeta1 stops a workflow using the beta1 spec.
+func (c *GRPCClient) TerminateWorkflowBeta1(ctx context.Context, req *TerminateWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to terminate workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.TerminateWorkflowBeta1(ctx, &pb.TerminateWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to terminate workflow: %v", err)
+	}
+	return nil
+}
+
+// PauseWorkflowBeta1 pauses a workflow that can be resumed later using the beta1 spec.
+func (c *GRPCClient) PauseWorkflowBeta1(ctx context.Context, req *PauseWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to pause workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.PauseWorkflowBeta1(ctx, &pb.PauseWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to pause workflow: %v", err)
+	}
+	return nil
+}
+
+// ResumeWorkflowBeta1 resumes a paused workflow using the beta1 spec.
+func (c *GRPCClient) ResumeWorkflowBeta1(ctx context.Context, req *ResumeWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to resume workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	_, err := c.protoClient.ResumeWorkflowBeta1(ctx, &pb.ResumeWorkflowRequest{
+		InstanceId:        req.InstanceID,
+		WorkflowComponent: req.WorkflowComponent,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to resume workflow: %v", err)
+	}
+	return nil
+}
+
+// RaiseEventWorkflowBeta1 raises an event on a workflow using the beta1 spec.
+func (c *GRPCClient) RaiseEventWorkflowBeta1(ctx context.Context, req *RaiseEventWorkflowRequest) error {
+	if req.InstanceID == "" {
+		return errors.New("failed to raise event on workflow: InstanceID must be supplied")
+	}
+	if req.WorkflowComponent == "" {
+		req.WorkflowComponent = DefaultWorkflowComponent
+	}
+	if req.EventName == "" {
+		return errors.New("failed to raise event on workflow: EventName must be supplied")
+	}
+	var eventData []byte
+	var err error
+	if req.SendRawData {
+		if reflect.ValueOf(req.EventData).Type() != typeBytes {
```

Comment:
This probably works too, but I thought it was more conventional in Go to do something like this:
```go
if eventData, ok = req.EventData.([]byte); !ok {
    return errors.New("...")
}
```

Reply:
Makes sense - implemented.